src/city/resource.c

```diff
@@ -58,6 +61,37 @@ int city_resource_get_amount_including_granaries(resource_type resource, int amo
     return amount_stored;
 }
 
+int city_resource_get_available_empty_space_granaries(resource_type food, int respect_settings)
+{
+    int available_storage = 0;
+    if (!resource_is_food(food)) {
+        return 0;
+    }
+
+    for (building *b = building_first_of_type(BUILDING_GRANARY); b; b = b->next_of_type) {
+        if (!respect_settings) {
+            available_storage += b->resources[RESOURCE_NONE];
+        } else {
+            available_storage += building_granary_maximum_receptible_amount(food, b);
+        }
+    }
+
+    return available_storage;
+}
+
+int city_resource_get_available_empty_space_warehouses(resource_type resource, int respect_settings)
+{
+    int available_storage = 0;
+    for (building *b = building_first_of_type(BUILDING_WAREHOUSE); b; b = b->next_of_type) {
+        if (!respect_settings) {
```

Comment:
Don't forget to check for valid warehouses.

Reply:
Done.